backend/metalit_challenge/api/views.py

```diff
@@ -0,0 +1,81 @@
+from django.shortcuts import render
+from rest_framework import serializers, status
+from rest_framework.decorators import action
+from rest_framework.views import APIView
+from rest_framework.viewsets import ViewSet
+from rest_framework.response import Response
+from .models import Challenge, Task, TaskVerification
+from .serializers import ChallengeSerializer, TaskSerializer, TaskVerificationSerializer
+
+# Create your views here.
+
+class ChallengeView(APIView):
+  def get(self, request):
+    """
+    GET all challenge already published
+    """
+    query = Challenge.objects.filter(status='published')
+    serializers = ChallengeSerializer(query, many=True)
+    return Response(serializers.data)
+
+class TaskView(APIView):
+  """
+  GET all tasks by challenge id
+  """
+  def get_task(self, challenge_id):
+    return Task.objects.filter(challenge_id=challenge_id)
+
+  def get(self, request, challenge_id):
+    query = self.get_task(challenge_id)
+    serializers = TaskSerializer(query, many=True)
+    if query.count() != 0:
+      return Response(serializers.data)
+    else:
+      return Response({"description": "challenge ID not found"}, status=status.HTTP_404_NOT_FOUND)
+
+class ChallengeTaskView(APIView):
+  """
+  GET challenge and all of it's tasks
+  """
+  def get_challenge(self, challenge_id):
+    return Challenge.objects.filter(id=challenge_id)
+
+  def get_task(self, challenge_id):
+    return Task.objects.filter(challenge_id=challenge_id)
+
+  def get(self, request, challenge_id):
+    challenge_query = self.get_challenge(challenge_id)
+    task_query = self.get_task(challenge_id)
+    challenge_serializers = ChallengeSerializer(challenge_query, many=True)
+    task_serializers = TaskSerializer(task_query, many=True)
+    if challenge_query.count() == 0 and task_query.count() == 0:
```

Comment:
@fadelananda If you only want to determine if at least one result exists (and don’t need the actual objects), it’s more efficient to use exists(). reference: https://docs.djangoproject.com/en/4.0/ref/models/querysets/#django.db.models.query.QuerySet.exists